community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/IndexingService.java

```diff
@@ -234,7 +235,16 @@ public void init()
                     break;
                 case POPULATING:
                     // The database was shut down during population, or a crash has occurred, or some other sad thing.
-                    indexProxy = proxySetup.createRecoveringIndexProxy( descriptor, providerDescriptor, constraint );
+                    if ( constraint && indexRule.getOwningConstraint() == null )
```

Comment:
The `IndexRule.getOwningConstraint()` compares a `Long` field value with a primitive `long` constant using `==`. It is uncomfortably not clear from the code if auto-boxing turns the object into a primitive to compare values, or the primitive gets boxed to compare references. It would calm my nerves if you could fix that up while you are already looking at index stuff.